### ISSUE TITLE
Error - zero-size array to reduction operation maximum which has no identity

### ISSUE THREAD

**Issue by annahedstroem** [sebastian-lapuschkin-sideprojects/voronoy-cells-toydata#7]
Ran this query:

```--random_seed 0xc0ffee  --size 224  --number 10000  --min_centroids 5  --max_centroids 10  --distance_metric chebyshev  --class_colors ./example_textures/0d2576a86f0f71df4e1dac9b1545a2bc.jpg ./example_textures/1f00b3d571e19632f4c5b7153912e50b.jpg ./example_textures/54580fea7ced5fe901ec8f9c017573dc.jpg  --class_color_deviation 40  --bg_colors ./example_textures/1ad90240a7ee520bdeddcda935d3710a.jpg  --bg_color_deviation 7  --marker_color class  --draw_borders color:0xffffff:linear:30  --line_dilation_iterations 1  --line_erosion_iterations 1  --output ./ds ```

and got this error (as 74% of the process had finished):

```
Traceback (most recent call last):
  File "main.py", line 185, in <module>
    rgb, canvas = fill_class_color(canvas, args.bg_colors[np.random.randint(low=0, high=len(args.bg_colors))], args.bg_color_deviation, (COORDS[I, 0], COORDS[I, 1]))
  File "main.py", line 98, in fill_class_color
    W_coord, H_coord = (np.max(coordinates[0]) - np.min(coordinates[0]), np.max(coordinates[1]) - np.min(coordinates[1]))
  File "<__array_function__ internals>", line 5, in amax
  File "/Users/annahedstroem/anaconda3/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 2705, in amax
    return _wrapreduction(a, np.maximum, 'max', axis, None, out,
  File "/Users/annahedstroem/anaconda3/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 87, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
ValueError: zero-size array to reduction operation maximum which has no identity
```

let's look into it! (running just after with 7000 samples worked).

**Comment by sebastian-lapuschkin** [comment]
fixed in f51d72d .

the problem appeared, when two or more centroids would be assigned exactly the same 2d-coordinates, and thus non-first duplicate centroids in the list would not be assigned any pixels.
an error occurred when trying to draw textures in an empty region.

fixed by removing non-unique centroids using `centroids = np.unique(centroids, axis=0)`
**CAUTION** previously generated datasets might not be reproducible anymore, since  `np.unique` might change the order of centroids along the specified axis.

**Comment by sebastian-lapuschkin** [comment]
reinstated compatibility to previously generated datasets in 82d7136
